Examples/Tutorials/Tutorial-DpdkL2Fwd/main.cpp

```diff
@@ -68,49 +52,46 @@ int main(int argc, char* argv[])
 	pcpp::DpdkDeviceList::initDpdk(coreMaskToUse, MBUF_POOL_SIZE);
 
 	// Find DPDK devices
-	pcpp::DpdkDevice* device1 = pcpp::DpdkDeviceList::getInstance().getDeviceByPort(DEVICE_ID_1);
-	if (device1 == NULL)
+	auto* device1 = pcpp::DpdkDeviceList::getInstance().getDeviceByPort(DEVICE_ID_1);
+	if (device1 == nullptr)
 	{
-		std::cerr << "Cannot find device1 with port '" << DEVICE_ID_1 << "'" << std::endl;
+		std::cerr << "Cannot find device1 with port '" << DEVICE_ID_1 << "'\n";
 		return 1;
 	}
 
-	pcpp::DpdkDevice* device2 = pcpp::DpdkDeviceList::getInstance().getDeviceByPort(DEVICE_ID_2);
-	if (device2 == NULL)
+	auto* device2 = pcpp::DpdkDeviceList::getInstance().getDeviceByPort(DEVICE_ID_2);
+	if (device2 == nullptr)
 	{
-		std::cerr << "Cannot find device2 with port '" << DEVICE_ID_2 << "'" << std::endl;
+		std::cerr << "Cannot find device2 with port '" << DEVICE_ID_2 << "'\n";
 		return 1;
 	}
 
 	// Open DPDK devices
 	if (!device1->openMultiQueues(1, 1))
 	{
-		std::cerr << "Couldn't open device1 #" << device1->getDeviceId() << ", PMD '" << device1->getPMDName() << "'" << std::endl;
+		std::cerr << "Couldn't open device1 #" << device1->getDeviceId() << ", PMD '" << device1->getPMDName() << "'\n";
 		return 1;
 	}
 
 	if (!device2->openMultiQueues(1, 1))
 	{
-		std::cerr << "Couldn't open device2 #" << device2->getDeviceId() << ", PMD '" << device2->getPMDName() << "'" << std::endl;
+		std::cerr << "Couldn't open device2 #" << device2->getDeviceId() << ", PMD '" << device2->getPMDName() << "'\n";
 		return 1;
 	}
 
 	// Create worker threads
 	std::vector<pcpp::DpdkWorkerThread*> workers;
-	workers.push_back(new L2FwdWorkerThread(device1, device2));
-	workers.push_back(new L2FwdWorkerThread(device2, device1));
+	// Constructs a DpdkWorkerThread* directly within the vector's storage
+	workers.emplace_back(new L2FwdWorkerThread(device1, device2));
+	workers.emplace_back(new L2FwdWorkerThread(device2, device1));
 
 	// Create core mask - use core 1 and 2 for the two threads
-	int workersCoreMask = 0;
-	for (int i = 1; i <= 2; i++)
-	{
-		workersCoreMask = workersCoreMask | (1 << i);
-	}
+	int workersCoreMask = 0x06; // Binary 110, using cores 1 and 2
```

Comment:
cleaner

Reply:
It is cleaner but I'm not sure it is more beginner friendly. The for-loop implementation shows that people might set mask by using bit shifting but the updated version only shows as a constant. Since this is a tutorial code, I'm not sure which one is better.